new windows pip-pre test failures

tuple-vs-list failures. https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=29144&view=logs&j=dded70eb-633c-5c42-e995-a7f8d1f99d91&t=d18f7f2f-13af-5901-1cbc-7fa039d0db3a&l=4001

```
FAILED mne/tests/test_epochs.py::test_to_data_frame_index[time] - AssertionError: assert ('time',) == ['time']
FAILED mne/tests/test_epochs.py::test_to_data_frame_index[index1] - AssertionError: assert ('condition', 'time', 'epoch') == ['condition', 'tim...
FAILED mne/tests/test_epochs.py::test_to_data_frame_index[index2] - AssertionError: assert ('epoch', 'time') == ['epoch', 'time']
FAILED mne/tests/test_epochs.py::test_to_data_frame_index[index3] - AssertionError: assert ('time', 'epoch') == ['time', 'epoch']
FAILED mne/tests/test_epochs.py::test_to_data_frame_index[None] - assert (None,) == [None]
FAILED mne/tests/test_source_estimate.py::test_to_data_frame_index[time] - AssertionError: assert ('time',) == ['time']
FAILED mne/tests/test_source_estimate.py::test_to_data_frame_index[index1] - AssertionError: assert ('time', 'subject') == ['time', 'subject']
FAILED mne/tests/test_source_estimate.py::test_to_data_frame_index[None] - assert (None,) == [None]
FAILED mne/time_frequency/tests/test_tfr.py::test_to_data_frame_index[time] - AssertionError: assert ('time',) == ['time']
FAILED mne/time_frequency/tests/test_tfr.py::test_to_data_frame_index[index1] - AssertionError: assert ('condition', 'time', 'freq') == ['condition', 'time...
FAILED mne/time_frequency/tests/test_tfr.py::test_to_data_frame_index[index2] - AssertionError: assert ('freq', 'time') == ['freq', 'time']
FAILED mne/time_frequency/tests/test_tfr.py::test_to_data_frame_index[index3] - AssertionError: assert ('time', 'freq') == ['time', 'freq']
FAILED mne/time_frequency/tests/test_tfr.py::test_to_data_frame_index[None] - assert (None,) == [None]
```

first appeared in #12431

--- a/mne/tests/test_epochs.py
+++ b/mne/tests/test_epochs.py
@@ -3086,7 +3086,7 @@ def test_to_data_frame_index(index):
     # test index order/hierarchy preservation
     if not isinstance(index, list):
         index = [index]
-    assert df.index.names == index
+    assert list(df.index.names) == index
     # test that non-indexed data were present as columns
     non_index = list(set(["condition", "time", "epoch"]) - set(index))
     if len(non_index):

--- a/mne/tests/test_source_estimate.py
+++ b/mne/tests/test_source_estimate.py
@@ -1240,7 +1240,7 @@ def test_to_data_frame_index(index):
     # test index setting
     if not isinstance(index, list):
         index = [index]
-    assert df.index.names == index
+    assert list(df.index.names) == index
     # test that non-indexed data were present as columns
     non_index = list(set(["time", "subject"]) - set(index))
     if len(non_index):

--- a/mne/time_frequency/tests/test_tfr.py
+++ b/mne/time_frequency/tests/test_tfr.py
@@ -1453,7 +1453,7 @@ def test_to_data_frame_index(index):
     # test index order/hierarchy preservation
     if not isinstance(index, list):
         index = [index]
-    assert df.index.names == index
+    assert list(df.index.names) == index
     # test that non-indexed data were present as columns
     non_index = list(set(["condition", "time", "freq", "epoch"]) - set(index))
     if len(non_index):

